src/Pulsar.Client/Internal/MultiTopicsConsumerImpl.fs

```diff
@@ -806,6 +807,15 @@ type internal MultiTopicsConsumerImpl<'T> (consumerConfig: ConsumerConfiguration
                 |> Seq.map (fun (KeyValue(_, (consumer, _))) -> consumer.LastDisconnectedTimestamp)
                 |> Seq.max
                 |> channel.SetResult
+            
+            | IsConnected channel ->
+
+                Log.Logger.LogDebug("{0} IsConnected", prefix)
+                consumers
+                |> Seq.map (fun (KeyValue(_, (consumer, _))) -> consumer.IsConnected)
+                |> Seq.contains(false)
+                |> not
```

Comment:
I think you can replace it with
`Seq.forall _.IsConnected`

Reply:
looks much better 😃